Add OpenStack-related documentation, incl. Cinder plugins

jsafrane opened the pull request:
The storage part is inspired by PR #924.

Cinder support is on it's way to Kubernetes, see https://github.com/kubernetes/kubernetes/pull/13367.
~~Do not merge until the code is in OpenShift!~~ Code has been merged to openshift/origin in commit 2009660eb84e1a56df68db8e1077e2b0bd136d3f.

adellape commented:
@jsafrane Is there an origin PR to follow yet? I didn't find one in my search, but just checking. Thanks for this!

jsafrane commented:
There is a Kubernetes PR https://github.com/kubernetes/kubernetes/pull/13367 (just merged yesterday), I think it will end up in OpenShift in the usual merge. I guess you know better than me when/how do you pull stuff from Kubernetes master.

jsafrane commented:
Rebased, added Fibre Channel guide by @rootfs.

jsafrane commented:
And looking at Kubernetes in current origin/master, I can see it already has Cinder volume plugin.

adellape commented on admin_guide/configuring_openstack.adoc:
```diff
@@ -0,0 +1,75 @@
+volumes as persistent storage for application data.
+
+=== Cloud configuration
+As an administrator, create *_/etc/cloud.conf_* file on all your OpenShift
```
s/create/create a/

jsafrane replied:
Fixed

adellape commented on admin_guide/configuring_openstack.adoc:
```diff
@@ -0,0 +1,75 @@
+----
+
+Consult your OpenStack administrators for values of `*OS_*` variables, they are
+commonly used in OpenStack world.
```
Suggestion:

"Consult your OpenStack administrators for values of the OS_ variables, which are commonly used in the OpenStack world."

jsafrane replied:
Fixed

adellape commented on admin_guide/configuring_openstack.adoc:
```diff
@@ -0,0 +1,75 @@
+=== Master configuration
+
+Edit or
+link:master_node_configuration#creating-new-configuration-files[create]
```
Target is missing `.html`. Should be:

`link:master_node_configuration.html#creating-new-configuration-files[create]`

jsafrane replied:
Fixed

adellape commented on admin_guide/configuring_openstack.adoc:
```diff
@@ -0,0 +1,75 @@
+=== Node configuration
+
+Edit or
+link:master_node_configuration#creating-new-configuration-files[create]
```
Same here re: missing `.html`.

jsafrane replied:
Fixed

adellape commented on admin_guide/persistent_storage/persistent_storage_cinder.adoc:
```diff
@@ -0,0 +1,85 @@
+
+For a detailed example, see the guide for
+https://github.com/openshift/origin/tree/master/examples/wordpress[WordPress and
+MySQL using persistent volumes].
```
Assuming carried over from the NFS topic. Probably not needed here since it's NFS-specific?

jsafrane replied:
I have OpenShift PR, which adds Cinder (and Fibre Channel): https://github.com/openshift/origin/pull/4403

adellape commented on admin_guide/persistent_storage/persistent_storage_cinder.adoc:
```diff
@@ -0,0 +1,85 @@
+
+This allows using unformatted Cinder volumes as persistent volumes, as OpenShift
+will format them before the first use. However, it can result in data loss, if
+value of `*fsType*` parameter is changed after the volume was already formatted
```
s/data loss, if value/data loss if the value/

jsafrane replied:
Fixed

adellape commented on admin_guide/persistent_storage/persistent_storage_cinder.adoc:
```diff
@@ -0,0 +1,85 @@
+This allows using unformatted Cinder volumes as persistent volumes, as OpenShift
+will format them before the first use. However, it can result in data loss, if
+value of `*fsType*` parameter is changed after the volume was already formatted
+and provissioned with data.
```
s/provissioned/provisioned/

jsafrane replied:
Fixed

adellape commented:
@jsafrane Some comments inline. Thanks!

adellape commented:
> Rebased, added Fibre Channel guide by @rootfs.

@jsafrane Oh, also, what did the above comment refer to?

jsafrane commented:
I fixed most of the review remarks.

>> Rebased, added Fibre Channel guide by @rootfs.

>@jsafrane Oh, also, what did the above comment refer to?

Sorry, that belongs to different PR. Too many open pull requests...

jsafrane commented:
And thanks for the review!

adellape commented:
@jsafrane Thanks, LGTM. Looks like it needs a rebase now, but it's probably just cuz of https://github.com/openshift/openshift-docs/pull/924 which I merged yesterday (introduces the `persistent_storage` subdir and Gluster topic). So it's likely just a matter of fixing the `_build_cfg.yaml` to make sure it includes all three (NFS, Gluster, Cinder).

jsafrane commented:
Rebased and squashed.

adellape commented:
:tada:

jsafrane commented:
Thanks!
